firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java

```diff
@@ -740,6 +798,47 @@ T deserialize(
           }
         }
       }
+
+      // Populate @DocumentId annotated fields. If there is a conflict (@DocumentId annotation is
+      // applied to a property that is already deserialized from the firestore document)
+      // a runtime exception will be thrown.
+      for (String docIdPropertyName : documentIdPropertyNames) {
```

Comment:
FWIW- This class has overly large methods, but typically we try to keep methods shorter.  Consider separating this out into a populateDocumentIdFields() method or something.

Reply:
Done.